Angaza Data Analysis Assignment

In [2]:
#Import all libraries to be used
import numpy as np
import pandas as pd 
import seaborn as sns
import matplotlib.pyplot as plt
import statsmodels.api as st
import plotly
plotly.offline.init_notebook_mode(connected=True)
import plotly.graph_objs as go
import plotly.express as px
from datetime import time, date, timedelta
from datetime import datetime
from dateutil import parser
import datetime as dt
import math
import dash

In [3]:
#Read data set
accts=pd.read_csv('accounts.csv')
group=pd.read_csv('groups.csv')
pymt=pd.read_csv('payments.csv')
ast=pd.read_csv('account_state_transitions.csv')
gpa=pd.read_csv('group_product_associations.csv')
products=pd.read_csv('products.csv')

display(accts.head())
display(group.head())
display(pymt.head())
display(ast.head())
display(gpa.head())
display(products.head())

display(accts.info())
display(group.info())
display(pymt.info())
display(ast.info())
display(gpa.info())
display(products.info())

id  group_id  organization_id           registration_date  is_written_off  \
0   2         1            10001  2019-11-06 05:49:39.571392           False   
1   3         1            10001  2019-11-14 14:57:02.571392           False   
2   4         1            10001  2019-12-11 03:18:53.571392           False   
3   5         1            10001  2019-10-31 22:36:23.571392           False   
4   6         1            10001  2019-09-29 13:04:45.571392           False   

   is_unlocked  
0         True  
1        False  
2         True  
3        False  
4         True

id                                          name  price_upfront  \
0   1                  Gugun Polar Wireless Product            950   
1   2                 Tennie JUA Home Mate H1G Lite           1500   
2   3  Hong Rath Omnivoltaic Catch (Keycode) Bundle            800   
3   4                        Monte Omnivoltaic M400            650   
4   5                 Coats Omnivoltaic M600 Bundle           1100   

   price_unlock  price_clock_hour  minimum_payment  
0          8100          5.654762             71.5  
1          6100          8.928571             46.0  
2          5200          4.761905             44.0  
3          6100          3.869048             54.5  
4          6100          6.547619             50.0

id              effective_when  account_id currency  amount
0   1  2019-12-26 06:00:42.571392           1      KES     950
1   2  2019-12-26 06:00:42.571392           1      KES     950
2   3  2019-11-06 05:49:39.571392           2      KES     950
3   4  2019-11-06 05:49:39.571392           2      KES     950
4   5  2019-11-11 06:43:34.571392           2      KES     415

id                created_when                started_when  account_id  \
0   3  2020-03-12 20:03:35.724574  2019-11-06 05:49:39.571392           2   
1   4  2020-03-12 20:03:35.724574  2019-11-25 22:20:59.150339           2   
2   5  2020-03-12 20:03:36.042456  2019-11-26 10:22:36.571392           2   
3   6  2020-03-12 20:03:36.042456  2019-11-29 11:03:32.150339           2   
4   7  2020-03-12 20:03:36.098448  2019-12-03 10:08:03.571392           2   

  from_state  to_state  previous_state_transition_id  
0    CREATED   ENABLED                           NaN  
1    ENABLED  DISABLED                           3.0  
2   DISABLED   ENABLED                           4.0  
3    ENABLED  DISABLED                           5.0  
4   DISABLED   ENABLED                           6.0

id  group_id  product_id
0   1         1          42
1   2         2          20
2   3         3          25
3   4         3          32
4   5         3          32

id               name                type
0   1        Bright Move  LARGEPADV1_KEYCODE
1   2  Bright Move Smart  LARGEPADV1_KEYCODE
2   3     Bright Home 29  LARGEPADV1_KEYCODE
3   4     Bright Home 58  LARGEPADV1_KEYCODE
4   5    Bright Sun Bell               CABLE

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 772 entries, 0 to 771
Data columns (total 6 columns):
id                   772 non-null int64
group_id             772 non-null int64
organization_id      772 non-null int64
registration_date    772 non-null object
is_written_off       772 non-null bool
is_unlocked          772 non-null bool
dtypes: bool(2), int64(3), object(1)
memory usage: 25.8+ KB


None

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 42 entries, 0 to 41
Data columns (total 6 columns):
id                  42 non-null int64
name                42 non-null object
price_upfront       42 non-null int64
price_unlock        42 non-null int64
price_clock_hour    42 non-null float64
minimum_payment     42 non-null float64
dtypes: float64(2), int64(3), object(1)
memory usage: 2.1+ KB


None

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9452 entries, 0 to 9451
Data columns (total 5 columns):
id                9452 non-null int64
effective_when    9452 non-null object
account_id        9452 non-null int64
currency          9452 non-null object
amount            9452 non-null int64
dtypes: int64(3), object(2)
memory usage: 369.3+ KB


None

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 14762 entries, 0 to 14761
Data columns (total 7 columns):
id                              14762 non-null int64
created_when                    14762 non-null object
started_when                    14762 non-null object
account_id                      14762 non-null int64
from_state                      14762 non-null object
to_state                        14762 non-null object
previous_state_transition_id    13990 non-null float64
dtypes: float64(1), int64(2), object(4)
memory usage: 807.4+ KB


None

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 107 entries, 0 to 106
Data columns (total 3 columns):
id            107 non-null int64
group_id      107 non-null int64
product_id    107 non-null int64
dtypes: int64(3)
memory usage: 2.6 KB


None

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 42 entries, 0 to 41
Data columns (total 3 columns):
id      42 non-null int64
name    42 non-null object
type    42 non-null object
dtypes: int64(1), object(2)
memory usage: 1.1+ KB


None

In [4]:
#Rename some columns for easier merging in exercises below
accts=accts.rename(columns={'id': 'account_id'})
group=group.rename(columns={'id': 'group_id', 'name': 'group_name'})
pymt=pymt.rename(columns={'id': 'pymt_id'})
ast=ast.rename(columns={'id': 'ast_id'})
gpa=gpa.rename(columns={'id': 'gpa_id'})
products=products.rename(columns={'id': 'product_id', 'name': 'product_name'})

Part 1: Data Handling:
1. Generate a bar chart visualizing the relative frequency of Products registered with Accounts.

In [11]:
dfq11 = accts.merge(group, on=['group_id']) \
    .merge(gpa, on=['group_id']) \
    .merge(products, on=['product_id'])
dfq11=dfq11.drop_duplicates(subset=['account_id','product_id']) \

display(dfq11.head())
dfq11.info()
dfq11.columns
dfq11.index

# Identify totals
num_prod = len(dfq11['product_name'].unique())
num_acct = len(dfq11['account_id'].unique())
tot_prod = len(dfq11)
print('Number of unique products = ', num_prod)
print('Number of unique accounts = ', num_acct)
print('Total products sold amongst all accounts = ', tot_prod)

# Count number of accts registered with each product and sort them in descending order
num_accts_in_prod = dfq11['product_name'].value_counts().sort_values(ascending = False)

data = [go.Bar(
        x = num_accts_in_prod.index,
        y = num_accts_in_prod.values)]

plotly.offline.iplot(data)

account_id  group_id  organization_id           registration_date  \
0           2         1            10001  2019-11-06 05:49:39.571392   
1           3         1            10001  2019-11-14 14:57:02.571392   
2           4         1            10001  2019-12-11 03:18:53.571392   
3           5         1            10001  2019-10-31 22:36:23.571392   
4           6         1            10001  2019-09-29 13:04:45.571392   

   is_written_off  is_unlocked                    group_name  price_upfront  \
0           False         True  Gugun Polar Wireless Product            950   
1           False        False  Gugun Polar Wireless Product            950   
2           False         True  Gugun Polar Wireless Product            950   
3           False        False  Gugun Polar Wireless Product            950   
4           False         True  Gugun Polar Wireless Product            950   

   price_unlock  price_clock_hour  minimum_payment  gpa_id  product_id  \
0          8100          5.654762             71.5       1          42   
1          8100          5.654762             71.5       1          42   
2          8100          5.654762             71.5       1          42   
3          8100          5.654762             71.5       1          42   
4          8100          5.654762             71.5       1          42   

             product_name     type  
0  Polar Wireless Product  IOT_API  
1  Polar Wireless Product  IOT_API  
2  Polar Wireless Product  IOT_API  
3  Polar Wireless Product  IOT_API  
4  Polar Wireless Product  IOT_API

<class 'pandas.core.frame.DataFrame'>
Int64Index: 889 entries, 0 to 1836
Data columns (total 15 columns):
account_id           889 non-null int64
group_id             889 non-null int64
organization_id      889 non-null int64
registration_date    889 non-null object
is_written_off       889 non-null bool
is_unlocked          889 non-null bool
group_name           889 non-null object
price_upfront        889 non-null int64
price_unlock         889 non-null int64
price_clock_hour     889 non-null float64
minimum_payment      889 non-null float64
gpa_id               889 non-null int64
product_id           889 non-null int64
product_name         889 non-null object
type                 889 non-null object
dtypes: bool(2), float64(2), int64(7), object(4)
memory usage: 99.0+ KB
Number of unique products =  42
Number of unique accounts =  772
Total products sold amongst all accounts =  889


Part 1: Data Handling:
2.  Devise a method of generating a table of all Accounts with their State as of any arbitrary timestamp in the past and the total time they had been in that state as of the timestamp.

In [16]:
timestamp2 = "2020-01-01 00:00:00"

dfq12 = ast.filter(['account_id','ast_id','started_when','to_state'])

dfq12['timestamp'] = timestamp2
dfq12['timestamp'] = pd.to_datetime(dfq12['timestamp'], format='%Y-%m-%d %H:%M:%S')
dfq12['started_when'] = pd.to_datetime(dfq12['started_when'], format='%Y-%m-%d %H:%M:%S')

dfq12['time_in_state'] = dfq12['timestamp'] - dfq12['started_when']

dfq12 = dfq12.loc[(dfq12['started_when'] <= timestamp2)]
dfq12 = dfq12.sort_values(['ast_id']).drop_duplicates('account_id', keep='last')
dfq12 = dfq12.sort_values('account_id')

dfq12 = dfq12.drop(columns=['ast_id', 'started_when', 'timestamp'])

display(dfq12.head(15))
display(dfq12.info())

account_id  to_state           time_in_state
11860           1   ENABLED  5 days 17:59:17.428608
12              2   ENABLED  0 days 23:17:54.428608
37              3   ENABLED  0 days 21:54:10.428608
51              4   ENABLED 20 days 20:41:06.428608
77              5  DISABLED  4 days 15:12:16.954924
111             6  DISABLED  2 days 20:46:02.533871
12321           7  DISABLED  8 days 05:41:42.060187
145             9   ENABLED  1 days 00:48:45.428608
164            10   ENABLED 16 days 20:21:28.428608
202            11  DISABLED  0 days 23:58:49.218082
239            12  DISABLED  5 days 05:08:54.428608
12917          13  DISABLED  4 days 15:31:25.228608
304            14  DISABLED  4 days 15:47:34.028608
327            15  DISABLED 60 days 15:31:02.428608
377            16  DISABLED  6 days 00:40:53.628608

<class 'pandas.core.frame.DataFrame'>
Int64Index: 470 entries, 11860 to 11762
Data columns (total 3 columns):
account_id       470 non-null int64
to_state         470 non-null object
time_in_state    470 non-null timedelta64[ns]
dtypes: int64(1), object(1), timedelta64[ns](1)
memory usage: 14.7+ KB


None

Part 1: Data Handling:
3.  Devise a method of generating a table of all Accounts with their cumulative payments to date as of any arbitrary timestamp in the past.

In [25]:
timestamp3 = "2020-01-01 00:00:00"

pymt['effective_when'] = pd.to_datetime(pymt['effective_when'], format='%Y-%m-%d %H:%M:%S')
pymtloc = pymt.loc[(pymt.effective_when <= timestamp3)]
dfq13 = pymtloc.groupby(['account_id'])['amount','effective_when'].sum()

display(dfq13.head(15))
display(dfq13.info())

amount
account_id        
1             1900
2             4998
3             4448
4             3501
5             4357
6             5740
7             3124
9             5563
10            3084
11            5826
12            4731
13            3783
14            5285
15            3224
16            5283

<class 'pandas.core.frame.DataFrame'>
Int64Index: 470 entries, 1 to 772
Data columns (total 1 columns):
amount    470 non-null int64
dtypes: int64(1)
memory usage: 7.3 KB


None

Part 1: Data Handling:
4. Devise a method of generating a table of all Accounts with their “nominal” expected payments as of any arbitrary timestamp in the past. In other words, we’re looking for the amount the distributor would have collected as of the test date assuming the client had paid the minimum payment instantaneously upon going to a “disabled” state until reaching the unlock price.

In [26]:
dfq14 = accts.merge(group, on=['group_id'])

timestamp4 = "2020-01-01 00:00:00"
dfq14['timestamp'] = timestamp4
dfq14['timestamp'] = pd.to_datetime(dfq14['timestamp'], format='%Y-%m-%d %H:%M:%S')
dfq14['registration_date'] = pd.to_datetime(dfq14['registration_date'], format='%Y-%m-%d %H:%M:%S')
dfq14 = dfq14.loc[(dfq14.registration_date <= timestamp4)]

dfq14['total_time'] = dfq14['timestamp'] - dfq14['registration_date']
dfq14['total_hours'] = dfq14['total_time'] / np.timedelta64(1,'h')

dfq14['hours_post_upfront_period'] = dfq14['total_hours'] - (dfq14['price_upfront']/dfq14['price_clock_hour'])

dfq14['hours_interval'] = dfq14['minimum_payment']/dfq14['price_clock_hour']

dfq14['interval_payments'] = dfq14['hours_post_upfront_period']/dfq14['hours_interval']
dfq14['interval_payments'] = ((dfq14['interval_payments']).astype(int).astype(float))

dfq14['expected_payments'] = dfq14['price_upfront'] + ((dfq14['interval_payments'])*dfq14['minimum_payment'])
dfq14['expected_payments'] = dfq14[['expected_payments','price_unlock']].min(axis=1)
dfq14['expected_payments'] = dfq14[['expected_payments','price_upfront']].max(axis=1)

dfq14 = dfq14[['account_id', 'expected_payments']].sort_values('account_id')

display(dfq14.head(15))
display(dfq14.info())

account_id  expected_payments
8            1              950.0
0            2             7528.0
1            3             6384.0
2            4             2809.0
3            5             8100.0
4            6             8100.0
9            7             6169.5
5            9             8100.0
6           10             2237.0
7           11             8100.0
11          12             6100.0
20          13             6100.0
12          14             6100.0
21          15             6100.0
13          16             6100.0

<class 'pandas.core.frame.DataFrame'>
Int64Index: 470 entries, 8 to 770
Data columns (total 2 columns):
account_id           470 non-null int64
expected_payments    470 non-null float64
dtypes: float64(1), int64(1)
memory usage: 11.0 KB


None

Question 5:  
How long does it typically take Account holders to repay the unlock price as a % of the nominal loan term?

Answer - 223%, more than double the original nominal loan term
only 292 out of 772 accounts were unlocked
%s subsetted per group_id also given below

In [27]:
dfq5 = accts.merge(group, on=['group_id']) \
    .merge(ast, on=['account_id'])

dfq5 = dfq5.loc[(dfq5.to_state == 'UNLOCKED')]
dfq5['registration_date'] = pd.to_datetime(dfq5['registration_date'], format='%Y-%m-%d %H:%M:%S')
dfq5['started_when'] = pd.to_datetime(dfq5['started_when'], format='%Y-%m-%d %H:%M:%S')

dfq5['actual_time_to_repay'] = dfq5['started_when'] - dfq5['registration_date']
dfq5['actual_time_to_repay'] = dfq5['actual_time_to_repay'] / np.timedelta64(1,'h')

dfq5['nominal_loan_term'] = (dfq5['price_unlock']) / (dfq5['price_clock_hour'])

dfq5['time_to_repay_as%_of_term'] = dfq5['actual_time_to_repay'] / dfq5['nominal_loan_term'] * 100

avg_pct = round(dfq5['time_to_repay_as%_of_term'].mean(),1)
print('Average time to repay =', round(avg_pct,1), '% of nominal loan term')

avg_pct_by_group = dfq5.groupby('group_id')['time_to_repay_as%_of_term'].mean()
print(avg_pct_by_group)

display(dfq5.head())
display(dfq5.info())


# Distribution by time% (each bin 50% of nominal)
data = [go.Histogram(
        x = dfq5['time_to_repay_as%_of_term']
)]

# Vertical dashed line to indicate the average 
layout = {'shapes': [{
              'type' :'line',
              'x0': avg_pct,
              'y0': 0,
              'x1': avg_pct,
              'y1': 80,
              'line': { 'dash': 'dashdot'}
          }]
          }

plotly.offline.iplot({'data': data, 'layout': layout})

Average time to repay = 223.2 % of nominal loan term
group_id
1     194.866030
2     679.718431
3     555.928584
4     295.398597
5     137.102486
6     328.428637
7     128.042106
8     175.197288
9     198.530045
10    289.637026
11    204.774802
12    119.281823
13    568.925645
14    308.255745
15    152.058047
16    303.136084
17    406.153163
18    118.255033
19    178.533993
20    324.439691
21    109.197955
22     81.800998
23    237.171923
24    167.653334
25    210.046816
26     13.810964
27     25.246177
28     23.940419
29    611.063909
30    419.594279
31    413.514450
32    122.468323
33    139.787810
34    245.769085
35    362.476255
36    196.200842
37    290.337683
38    566.981481
39     16.420016
40     36.357399
41    217.034024
42    236.050316
Name: time_to_repay_as%_of_term, dtype: float64


account_id  group_id  organization_id          registration_date  \
28            2         1            10001 2019-11-06 05:49:39.571392   
67            4         1            10001 2019-12-11 03:18:53.571392   
126           6         1            10001 2019-09-29 13:04:45.571392   
163           9         1            10001 2019-10-10 12:49:10.571392   
176          10         1            10001 2019-12-15 03:38:31.571392   

     is_written_off  is_unlocked                    group_name  price_upfront  \
28            False         True  Gugun Polar Wireless Product            950   
67            False         True  Gugun Polar Wireless Product            950   
126           False         True  Gugun Polar Wireless Product            950   
163           False         True  Gugun Polar Wireless Product            950   
176           False         True  Gugun Polar Wireless Product            950   

     price_unlock  price_clock_hour  minimum_payment  ast_id  \
28           8100          5.654762             71.5      31   
67           8100          5.654762             71.5     258   
126          8100          5.654762             71.5     483   
163          8100          5.654762             71.5     680   
176          8100          5.654762             71.5     693   

                   created_when               started_when from_state  \
28   2020-03-12 20:03:37.120446 2020-02-24 02:00:21.571392   DISABLED   
67   2020-03-12 20:03:40.459370 2020-02-25 04:28:59.571392   DISABLED   
126  2020-03-12 20:03:44.402369 2020-02-20 18:54:47.571392   DISABLED   
163  2020-03-12 20:03:49.246836 2020-03-04 07:45:50.571392   DISABLED   
176  2020-03-12 20:03:50.585713 2020-02-21 15:48:49.571392   DISABLED   

     to_state  previous_state_transition_id  actual_time_to_repay  \
28   UNLOCKED                          30.0           2636.178333   
67   UNLOCKED                         257.0           1825.168333   
126  UNLOCKED                         482.0           3461.833889   
163  UNLOCKED                         679.0           3498.944444   
176  UNLOCKED                         692.0           1644.171667   

     nominal_loan_term  time_to_repay_as%_of_term  
28         1432.421053                 184.036553  
67         1432.421053                 127.418424  
126        1432.421053                 241.677116  
163        1432.421053                 244.267873  
176        1432.421053                 114.782707

<class 'pandas.core.frame.DataFrame'>
Int64Index: 292 entries, 28 to 14673
Data columns (total 20 columns):
account_id                      292 non-null int64
group_id                        292 non-null int64
organization_id                 292 non-null int64
registration_date               292 non-null datetime64[ns]
is_written_off                  292 non-null bool
is_unlocked                     292 non-null bool
group_name                      292 non-null object
price_upfront                   292 non-null int64
price_unlock                    292 non-null int64
price_clock_hour                292 non-null float64
minimum_payment                 292 non-null float64
ast_id                          292 non-null int64
created_when                    292 non-null object
started_when                    292 non-null datetime64[ns]
from_state                      292 non-null object
to_state                        292 non-null object
previous_state_transition_id    292 non-null float64

None

Question 6:
What % of the total loan value is typically paid at 100% of the nominal term?    

Answer - 67.1% of loan value on average is paid by nominal term
        

In [7]:
dfq6 = accts.merge(group, on=['group_id']) \
    .merge(pymt, on=['account_id'])

dfq6 = pd.concat([dfq6, ast['created_when']], axis=1)


dfq6['registration_date'] = pd.to_datetime(dfq6['registration_date'], format='%Y-%m-%d %H:%M:%S')
dfq6['effective_when'] = pd.to_datetime(dfq6['effective_when'], format='%Y-%m-%d %H:%M:%S')
dfq6['created_when'] = pd.to_datetime(dfq6['created_when'], format='%Y-%m-%d %H:%M:%S')

dfq6['nominal_loan_term'] = (dfq6['price_unlock']) / (dfq6['price_clock_hour'])
dfq6['nominal_loan_term_maturity'] = pd.to_datetime(dfq6['registration_date']) + pd.to_timedelta(dfq6['nominal_loan_term'],unit='h')

dfq6['cum_amount'] = dfq6.groupby(['account_id'])['amount'].cumsum()
dfq6['%_paid'] = dfq6['cum_amount'] / dfq6['price_unlock'] * 100

dfq6 = dfq6.loc[(dfq6.created_when >= dfq6.nominal_loan_term_maturity)]
dfq6 = dfq6.loc[(dfq6.effective_when <= dfq6.nominal_loan_term_maturity)]

dfq6 = dfq6.sort_values(['account_id','effective_when']).drop_duplicates('account_id', keep='last')


avg_pct_loan = dfq6['%_paid'].mean() 
print('Average % of total loan value paid at end of nominal term =', round(avg_pct_loan,1), '%')


display(dfq6.head())
display(dfq6.info())

# Distribution by time% (each bin 5% increment of nominal)
data = [go.Histogram(
        x = dfq6['%_paid']
)]

# Vertical dashed line to indicate the average 
layout = {'shapes': [{
              'type' :'line',
              'x0': avg_pct_loan,
              'y0': 0,
              'x1': avg_pct_loan,
              'y1': 100,
              'line': { 'dash': 'dashdot'}
          }]
          }

plotly.offline.iplot({'data': data, 'layout': layout})

Average % of total loan value paid at end of nominal term = 67.1 %


account_id  group_id  organization_id          registration_date  \
144         1.0       1.0          10001.0 2019-12-26 06:00:42.571392   
9           2.0       1.0          10001.0 2019-11-06 05:49:39.571392   
26          3.0       1.0          10001.0 2019-11-14 14:57:02.571392   
43          4.0       1.0          10001.0 2019-12-11 03:18:53.571392   
54          5.0       1.0          10001.0 2019-10-31 22:36:23.571392   

    is_written_off is_unlocked                    group_name  price_upfront  \
144          False       False  Gugun Polar Wireless Product          950.0   
9            False        True  Gugun Polar Wireless Product          950.0   
26           False       False  Gugun Polar Wireless Product          950.0   
43           False        True  Gugun Polar Wireless Product          950.0   
54           False       False  Gugun Polar Wireless Product          950.0   

     price_unlock  price_clock_hour  minimum_payment  pymt_id  \
144        8100.0          5.654762             71.5      2.0   
9          8100.0          5.654762             71.5     12.0   
26         8100.0          5.654762             71.5     29.0   
43         8100.0          5.654762             71.5     46.0   
54         8100.0          5.654762             71.5     57.0   

                effective_when currency  amount               created_when  \
144 2019-12-26 06:00:42.571392      KES   950.0 2020-03-12 20:03:47.925114   
9   2019-12-31 00:42:05.571392      KES   395.0 2020-03-12 20:03:36.247861   
26  2020-01-08 07:02:59.571392      KES   338.0 2020-03-12 20:03:37.007766   
43  2020-02-02 16:59:13.571392      KES   530.0 2020-03-12 20:03:38.848432   
54  2019-12-24 17:39:45.571392      KES   357.0 2020-03-12 20:03:39.952277   

     nominal_loan_term    nominal_loan_term_maturity  cum_amount     %_paid  
144        1432.421053 2020-02-23 22:25:58.360647600      1900.0  23.456790  
9          1432.421053 2020-01-04 22:14:55.360647600      4998.0  61.703704  
26         1432.421053 2020-01-13 07:22:18.360647600      5113.0  63.123457  
43         1432.421053 2020-02-08 19:44:09.360647600      6364.0  78.567901  
54         1432.421053 2019-12-30 15:01:39.360647600      4643.0  57.320988

<class 'pandas.core.frame.DataFrame'>
Int64Index: 518 entries, 144 to 9416
Data columns (total 20 columns):
account_id                    518 non-null float64
group_id                      518 non-null float64
organization_id               518 non-null float64
registration_date             518 non-null datetime64[ns]
is_written_off                518 non-null object
is_unlocked                   518 non-null object
group_name                    518 non-null object
price_upfront                 518 non-null float64
price_unlock                  518 non-null float64
price_clock_hour              518 non-null float64
minimum_payment               518 non-null float64
pymt_id                       518 non-null float64
effective_when                518 non-null datetime64[ns]
currency                      518 non-null object
amount                        518 non-null float64
created_when                  518 non-null datetime64[ns]
nominal_loan_term             518 non-null float64
nominal_

None